core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -383,15 +384,22 @@ private void processData(CohortDetails cohortDetails,
 				}
 				// if user specifies patients, add these to hashset, and send to
 				// GetMutationData
-				else if (sampleIds != null && sampleIds.length()>0) {
-                    // TODO: temporary solution
-				    String[] _keys = sampleIds.split("\\+");
-                    List<String> _arr = new ArrayList<String>();
-                    for (String _key: _keys) {
-                        String[] _tmp = _key.split("\\|");
-                        _arr.add(_tmp[1]);
+				else if (sampleIdsStr != null && sampleIdsStr.length() > 0) {
+                    List<String> sampleIds = new ArrayList<>();
+				    if (sampleIdsStr.contains("|")) { //TODO: a better way to distinguish input from "Case Ids" box or "Summary" page
```

Comment:
yes. The format of samples+studies from the summary page is different from the main query form. There has been a problem using the same format. I marked TODO for this so to find a better solution in the future.